### ISSUE TITLE
`scatter` and `plot` don't plot multiple sets of points

### ISSUE THREAD

**Issue by choldgraf** [maartenbreddels/ipyvolume#62]
In the docs it suggests you could do something like:

```
data = np.random.randn(3, 1000, 100)
ipyv.plot(*data)
```
and that this would plot 1000 lines of length 100.

Currently when I run the above code, it only plots a single line (as if I had run `ipyv.plot(*data[:, 0, :])`. Any idea what's going on?

**Comment by maartenbreddels** [comment]
Yes, there are 1000 lines, but not plotted at once, but for use in animations:
```python
data = np.random.randn(3, 1000, 100)
ipv.figure()
line = ipv.plot(*data)
ipv.animation_control(line)
ipv.show()
```
Or manually setting `line.sequence_index = 44`, for 'keyframe' 44.
I guess this needs some clarification in the api docs?

**Comment by choldgraf** [comment]
ooohhhhhhhhh that does make sense. Yeah that should be clarified :-) I'll make a PR this afternoon

so the expected way to plot multiple lines is to use a for-loop then? Are we taking any big performance hits?

@arokem wants to make beautiful plots like these:

![image](https://user-images.githubusercontent.com/1839645/30118876-f736f87a-92d8-11e7-9c48-e228c6afe889.png)

**Comment by maartenbreddels** [comment]
Hi Chris,

They are beautiful indeed! I'd say a better way is to do this is to use the plot_trisurf, which uses the Mesh widget.
The problem with lines in Open/Web GL is that you either draw a strip (where you connect all the subsequent vertices, or rather indices to the vertices) or a series of lines (from vertex index i, to j, then from k to l, ...) and cannot say draw from index i to j and then to k, and then one from l to m.
So to get the best performance (I think) is to draw this using a series of lines, lets take an example code.
```
ipv.figure()
x = np.arange(5) 
y = x**0.5
z = x**2
# draws two lines segments, one from 0->1->2 and one from 3->4
ipv.plot_trisurf(x, y, z, lines=np.array([0,1,1,2,3,4]))
ipv.show()
```
![lines](https://user-images.githubusercontent.com/1765949/30153294-6b0e1764-93b6-11e7-9b2e-9dc9acb34bb5.png)

Here the lines reference the vertex index, so no vertex (x,y,z) position will be duplicated.
If you have ideas for a better API for this, or know an existing API that we may mimic, let me know!
Can I have the data or code to make these plots?

cheers,

Maarten

**Comment by choldgraf** [comment]
Nice - we figured out a similar hack to basically concatenate lines with `np.nan`s in between. It achieves the same effect. Maybe worth writing an example around this?

**Comment by maartenbreddels** [comment]
That is a nice hack actually!
Do you put in a vertex with coordinates (NaN, NaN, NaN), or do you use NaN for the index? Since indices will be converted to int32, which will result in `-2147483648`. Either way it is a hack, so before taking this as a good solution I'd have to dig in the specs if this is expected behaviour.
I'll keep this open until at least we get this in the API, any strong opinions about it? The input format should I guess be a list of ndarrays, sth like this:
```python
ipv.plot_lines(list_of_x_arrays, list_of_y_arrays, list_of_z_arrays)
```

**Comment by vidartf** [comment]
You've probably stumbled upon the *primitive restart* functionality.

> GL_PRIMITIVE_RESTART_FIXED_INDEX
    Enables primitive restarting. If enabled, any one of the draw commands which transfers a set of generic attribute array elements to the GL will restart the primitive when the index of the vertex is equal to 2^n−1 where n is 8, 16 or 32 if the type is GL_UNSIGNED_BYTE, GL_UNSIGNED_SHORT, or GL_UNSIGNED_INT, respectively.

**Comment by vidartf** [comment]
Although that requires Webgl 2.0 and using vertex indices, so it should probably not be taken as according to spec.

**Comment by choldgraf** [comment]
we don't use vertices etc for this, since it's just plotting lines and not trisurf...are you talking about using the same hack for trisurf stuff?

**Comment by vidartf** [comment]
I'm confused. How do you plot the lines? The normal OpenGL way of drawing lines is to pass it a set of vertices (vertex buffer) to draw the lines between. This is the standard three.js way of drawing lines as well.

As a side-note, GL primitives are either points, lines, or surfaces (~polygons).

**Comment by maartenbreddels** [comment]
So `ipv.plot()` will use the Scatter widget, which will use the `gl.LINE_STRIP` method which doesn't require indices, adding a NaN there may indeed work (so what I said above is nonsense). Using trisurf is going through the Mesh widget, which is using indices and `gl.LINES`.

**Comment by maartenbreddels** [comment]
For the record, see this PR https://github.com/satra/niwidgets/pull/1 and https://github.com/maartenbreddels/ipyvolume/issues/71 for how this was done efficiently using the Mesh. May be useful to put this into a function, or an example.

**Comment by maartenbreddels** [comment]
Keeping open as a good first time issue, add in sth like `ipv.plot_multiline(...)`
